pkg/ipam/crd_eni.go

```diff
@@ -104,6 +106,13 @@ func setupENIDevices(eniConfigByMac configMap) {
 				Error("Failed to configure ENI device")
 		}
 	}
+
+	// Attach relevant BPF program to them.
+	if len(eniLinkByMac) > 0 {
+		if err := loader.ReinitializeIPsec(context.TODO()); err != nil {
```

Comment:
There is one thing I just remembered which might be relevant: Because `setupENIDevices` runs in a go routine in the background, there might be multiple concurrent invocations of it.

So `ReinitializeIPsec` must be able to deal with the fact that it might be called concurrently. Which looking at the implementation, I don't think it is, especially the synchronized write to `option.Config.EncryptInterface` looks like a potential data race to me.

(Another reason why we should refactor this whole logic here :disappointed: )


Also, feel free to introduce a context at the top of `setupENIDevices` like this:
```
func setupENIDevices(eniConfigByMac configMap) {
	ctx, cancel = context.WithCancel(context.Background())
	defer cancel()
```

Not sure why `setupENIDevices` did not use a context for e.g. the timeout, probably because it was implemented rather quickly to address a bug fix.